python/sglang/srt/managers/router/model_rpc.py

```diff
@@ -414,6 +413,19 @@ def forward_fill_batch(self, batch: Batch):
             if logprobs is not None:
                 req.logprob = logprobs[pt : pt + req.extend_input_len - 1]
                 req.normalized_logprob = normalized_logprobs[i]
+
+                # Cannot directly take logprobs[pt + req.extend_input_len - 1]
+                # as it is always the logprob of token 0. This is because at the
+                # time when logprobs is computed, the next token ID is not yet sampled.
+                assert logits is not None
+                token_ids = req.input_ids + [next_token_ids[i]]
+
+                next_token_logprobs = torch.log(torch.softmax(logits[i].float(), dim=-1) + 1e-6)
+                next_token_logprobs = next_token_logprobs.cpu().tolist()
+                token_texts = req.tokenizer.convert_ids_to_tokens(token_ids)
```

Comment:
Try to move all tokenization/detokenization into `detokenizer_manager.py`, so we can parallelize them

Reply:
Done. Also move this logic to OpenAI API frontend, so that SGLang frontend won't have this overhead.